scripts/library.sh

```diff
@@ -235,26 +235,52 @@ function report_go_test() {
   echo "Finished run, return code is ${failed}"
   # Tests didn't run.
   [[ ! -s ${report} ]] && return 1
-  # Create WORKSPACE file, required to use bazel
-  touch WORKSPACE
+  # Create WORKSPACE file, required to use bazel, if necessary
+  local has_workspace=0
+  if [[ -e WORKSPACE ]]; then
+    has_workspace=1
+  else
+    touch WORKSPACE
+  fi
   local targets=""
+  local last_run=""
+  local test_files=""
   # Parse the report and generate fake tests for each passing/failing test.
   echo "Start parsing results, summary:"
```

Comment:
Can we create another function to parse the results? This seems to be a lot of stuff and not easier to read :)

Reply:
That was my original intention, but it revealed to be non trivial and I don't want to spend cycles on this now.